src/TraceEvent/TraceLog.cs

```diff
@@ -4287,13 +4462,15 @@ public override bool Process()
                         TraceLog.rawKernelEventSource.Process();
                         TraceLog.rawEventSourceToConvert.StopProcessing();
                     });
-                    kernelTask.Start();
                 }
                 TraceLog.rawEventSourceToConvert.Process();
                 if (kernelTask != null)
                 {
                     TraceLog.rawKernelEventSource.StopProcessing();
                     kernelTask.Wait();
+
+                    // Flush all outstanding events in the realTimeQueue.
+                    TraceLog.FlushRealTimeEvents();
```

Comment:
Can you describe the reason for this?  Just want to make sure I understand if this is due to a bug or design change.

Reply:
I'd say it was a bug here - after Process() exited, you wouldn't have received all the events yet because the realtime queue has a flush interval.

I've noticed this issue at a certain time when the EventPipe was also routed through this queue. On the other hand, it doesn't use it anymore so the fix here isn't really helping **me**, but it still is a fix, although unrelated to the changes in this PR...